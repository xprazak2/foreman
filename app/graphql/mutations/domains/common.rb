module Mutations
  module Domains
    module Common
      extend ActiveSupport::Concern

      included do
        argument :name, String, required: false
        argument :fullname, String, required: false
        argument :dns_id, GraphQL::Types::ID, loads: Types::SmartProxy, required: false

        include ::Mutations::Concerns::TaxonomyArgs

        field :domain, Types::Domain, 'A domain.', null: true
      end
    end
  end
end
