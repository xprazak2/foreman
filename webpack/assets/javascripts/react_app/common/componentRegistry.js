import React from 'react';
import store from '../redux';
import PieChart from '../components/common/charts/PieChart/';
import StatisticsChartsList from '../components/statistics/StatisticsChartsList';
import PowerStatus from '../components/hosts/powerStatus/';
import NotificationContainer from '../components/notifications/';
import ToastsList from '../components/toastNotifications/';
import StorageContainer from '../components/hosts/storage/vmware/';
import _ from 'lodash';

let componentRegistry = () => {
    let registry = {
      PieChart: {
        type: PieChart,
        store: true,
        data: true
      },
      StatisticsChartsList: {
        type: StatisticsChartsList,
        store: true,
        data: true
      },
      PowerStatus: {
        type: PowerStatus,
        store: true,
        data: true
      },
      NotificationContainer: {
        type: NotificationContainer,
        store: true,
        data: true
      },
      ToastNotifications: {
        type: ToastsList,
        store: true,
        data: false
      }
    };

    return function () {
      let register = (name, component) => {
        if (registry[name]) {
          throw 'Component name already taken';
        }
        registry[name] = component;
        console.log('Component registered with name ' + name);
        return registry;
      };

      let getComponent = (name) => {
        return registry[name];
      };

      let registeredComponents = () => {
        return _.map(registry, (value, key) => {
          return key;
        }).join(', ');
      };

      let markup = (name, data, store) => {
        let currentComponent = getComponent(name);
        if (!currentComponent) {
          throw `Component not found:  ${name} among ${registeredComponents()}`
        }
        let ComponentName = currentComponent.type;
        return <ComponentName data={ currentComponent.data ? data : undefined } store={ currentComponent.store ? store : undefined } />;
      };

      return { register, getComponent, markup };
    }();
};

export default componentRegistry();
