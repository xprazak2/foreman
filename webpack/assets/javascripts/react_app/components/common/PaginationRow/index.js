import PaginationRow from './PaginationRow';

export default PaginationRow;
