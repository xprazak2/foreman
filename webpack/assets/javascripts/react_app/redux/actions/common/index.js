import API from '../../../API';

export const ajaxRequestAction = ({
  dispatch,
  requestAction,
  successAction,
  failedAction,
  url,
  item,
  headers = {},
  params = {},
}) => {
  dispatch({ type: requestAction, payload: item });
  return API.get(url, headers, params)
    .then(({ data }) => dispatch({ type: successAction, payload: { ...item, ...data } }))
    .catch(error => dispatch({ type: failedAction, payload: { error, item } }));
};
