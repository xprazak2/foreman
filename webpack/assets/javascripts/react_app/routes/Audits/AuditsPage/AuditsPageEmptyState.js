import React from 'react';
import PropTypes from 'prop-types';
import DefaultEmptyState from '../../../components/common/EmptyState';

const AuditsPageEmptyState = ({ message }) => (
  <DefaultEmptyState
    icon={message.type === 'error' ? 'error-circle-o' : 'add-circle-o'}
    header={message.type === 'error' ? __('Error') : __('No Audits Found')}
    description={message.text}
  />
);

AuditsPageEmptyState.propTypes = {
  message: PropTypes.shape({
    type: PropTypes.string,
    text: PropTypes.string,
  }).isRequired,
};

export default AuditsPageEmptyState;
