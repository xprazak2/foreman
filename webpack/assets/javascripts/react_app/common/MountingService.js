import React from 'react';

// import StorageContainer from '../components/hosts/storage/vmware/';
// import StatisticsChartsList from '../components/statistics/StatisticsChartsList';
// import PowerStatus from '../components/hosts/powerStatus/';
// import NotificationContainer from '../components/notifications/';
// import ToastsList from '../components/toastNotifications/';
// import PieChart from '../components/common/charts/PieChart/';
import ReactDOM from 'react-dom';
import store from '../redux';
import componentRegistry from './componentRegistry';

export function mount(component, selector, data) {
  // const components = {
  //   PieChart: {
  //     type: PieChart,
  //     markup: <PieChart data={ data } />
  //   },
  //   StatisticsChartsList: {
  //     type: StatisticsChartsList,
  //     markup: <StatisticsChartsList store={store} data={data}/>
  //   },
  //   PowerStatus: {
  //     type: PowerStatus,
  //     markup: <PowerStatus store={store} data={data}/>
  //   },
  //   NotificationContainer: {
  //     type: NotificationContainer,
  //     markup: <NotificationContainer store={store} data={data} />
  //   },
  //   ToastNotifications: {
  //     type: ToastsList,
  //     markup: <ToastsList store={store} />
  //   },
  //   StorageContainer: {
  //     type: StorageContainer,
  //     markup: <StorageContainer store={store} data={data} />
  //   }
  // };
  const reactNode = document.querySelector(selector);

  if (reactNode) {
    ReactDOM.unmountComponentAtNode(reactNode);
    ReactDOM.render(componentRegistry.markup(component, data, store), reactNode);
  } else {
    const componentName = componentRegistry(component).type.name;

    // eslint-disable-next-line no-console
    console.log(`Cannot find \'${selector}\' element for mounting the \'${componentName}\'`);
  }
}
