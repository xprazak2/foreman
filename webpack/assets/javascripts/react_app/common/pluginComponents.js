import componentRegistry from './componentRegistry';
import _ from 'lodash';

export default function register(bundleNames) {
  console.log(bundleNames)
  _.forEach(bundleNames, (name) => {
    let pluginComponents = require(name + '/components');
    registerPluginComponents(pluginComponents);
  });
}

let registerPluginComponents = (pluginComponents) => {
  for (let [name, component] of Object.entries(pluginComponents)) {
    componentRegistry.register(name, component);
  }
};
