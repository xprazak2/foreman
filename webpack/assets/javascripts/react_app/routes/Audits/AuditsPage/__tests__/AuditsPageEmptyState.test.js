import { testComponentSnapshotsWithFixtures } from '../../../../common/testHelpers';

import AuditsPageEmptyState from '../AuditsPageEmptyState';

const fixtures = {
  'render AuditsPageEmptyState w/error': {
    message: { type: 'error', text: 'error' },
  },
  'render AuditsPageEmptyState w/empty': {
    message: { type: 'empty', text: 'empty' },
  },
};

describe('AuditsPageEmptyState', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(AuditsPageEmptyState, fixtures));
});
