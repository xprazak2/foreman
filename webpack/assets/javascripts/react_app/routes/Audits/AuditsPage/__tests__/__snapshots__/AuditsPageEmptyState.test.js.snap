// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditsPageEmptyState rendering render AuditsPageEmptyState w/empty 1`] = `
<DefaultEmptyState
  description="empty"
  header="No Audits Found"
  icon="add-circle-o"
  iconType="pf"
  secondaryActions={Array []}
/>
`;

exports[`AuditsPageEmptyState rendering render AuditsPageEmptyState w/error 1`] = `
<DefaultEmptyState
  description="error"
  header="Error"
  icon="error-circle-o"
  iconType="pf"
  secondaryActions={Array []}
/>
`;
